ScheduleEvaluator/ConcreteCriterias/CoreClassesLastYear.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ScheduleEvaluator.ConcreteCriterias
+{
+    class CoreClassesLastYear : Criteria
+    {
+        public CoreClassesLastYear(double weight) : base(weight)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override double getResult(ScheduleModel s)
+        {
+            long lastYear = -1;
+            foreach (Quarter q in s.Quarters) {
+                if (q.Year > lastYear) {
+                    lastYear = q.Year;
+                }
+            }
+            double sum = 0;
+            foreach (Quarter q in s.Quarters) {
+                if (q.Year == lastYear) {
+                    foreach (Course c in q.Courses) {
+                        if (c.DepartmentID == s.PreferenceSet.DepartmentID) {
+                            sum++;
+                        }
+                    }
+                }
+            }
+            return sum;
```

Comment:
wait why are we returning sum? We just need to check that there's 1 core course in the last year right? In that case, we should reutrn max rating instead of the sum. If max rating is 3, and we return a sum > 3, the rating when added to other criteria will not be weighted correctly.